### ISSUE TITLE
Tutorial: Write a custom string interpolation handler – couple of small mistakes

### ISSUE THREAD

**Issue by Uzivatel919** [dotnet/docs#27209]
### # 1
```csharp
[InterpolatedStringHandler]
public ref struct LogInterpolatedStringHandler
{
    …
    // Add the receiver argument:
    …
```
I guess `// Add the receiver argument:` is forgotten since [Improved Interpolated Strings](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.defaultinterpolatedstringhandler?view=net-6.0) discusses declaratively _receiver_ term. 

### # 2
```csharp
public LogInterpolatedStringHandler(int literalLength, int formattedCount, LogLevel level, Logger logger, out bool isEnabled)
{
    enabled = logger.EnabledLevel >= level;
    Console.WriteLine($"\tliteral length: {literalLength}, formattedCount: {formattedCount}");
    if (enabled)
    {
        builder = new StringBuilder(literalLength);
    }
    isEnabled = enabled;
}
```

In fact there are 2 failures:
* `logger` and `level` parameters are switched so declaration `[InterpolatedStringHandlerArgument("", "level")] LogInterpolatedStringHandler builder` does not work with it.
* `LogInterpolatedStringHandler` is declared as `struct` so `builder` must be definitely assigned when .ctor returns. So either it must be changed to `class` or used something like `builder = enabled ? new () : null;`.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 99599c47-14ba-6e35-9b63-daa928f764bb
* Version Independent ID: e80a06db-bfbc-29f1-1cca-248460e523d3
* Content: [Explore string interpolation handlers](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/improved-interpolated-strings)
* Content Source: [docs/csharp/whats-new/tutorials/interpolated-string-handler.md](https://github.com/dotnet/docs/blob/main/docs/csharp/whats-new/tutorials/interpolated-string-handler.md)
* Product: **dotnet-csharp**
* Technology: **csharp-whats-new**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

**Comment by BillWagner** [comment]
I'll leave this open to inviestigate.

For #1, that's a stray comment in the initial sample and should be removed.

For #2, the code compiles in the samples repo, so I'll have to look at where the text disagrees with the sample code.

**Comment by Uzivatel919** [comment]
> For #2, the code compiles in the samples repo, so I'll have to look at where the text disagrees with the sample code.

Yes, it gets compiled since call site in snippets specified correctly `[InterpolatedStringHandlerArgument("level", "")] LogInterpolatedStringHandler builder)` but this implementation is not used on page.
